dags/common/enhancer.py

```diff
@@ -0,0 +1,52 @@
+import datetime
+
+
+class Enhancer:
+    def __constuct_abstracts(self, item, publisher):
+        item["abstracts"] = [{"value": item.pop("abstracts", ""), "source": publisher}]
+
+    def __construct_acquisition_source(self, item, publisher):
+        item["acquisition_source"] = {
+            "source": publisher,
+            "method": publisher,
+            "date": self.creation_date,
+            "submission_number": "path/to/the/file",
+        }
+
+    def __construct_copyright(self, item):
+        item["copyright"] = [
+            {
+                "holder": item.pop("copyright_holder", ""),
+                "year": item.pop("copyright_year", ""),
+                "statement": item.pop("copyright_statement", ""),
+                "material": item.pop("copyright_material", ""),
+            }
+        ]
+
+    def __construct_imprints(self, item, publisher):
+        item["imprints"] = [
+            {"date": item.pop("date_published", ""), "publisher": publisher}
+        ]
+
+    def __construct_record_creation_date(self, item):
+        item["record_creation_date"] = self.creation_date
+
+    def __construct_titles(self, item, publisher):
+        item["titles"] = [
+            {
+                "title": item.pop("title", ""),
+                "subtitle": item.pop("subtitle", ""),
+                "source": publisher,
+            }
+        ]
+
+    def __call__(self, publisher, item):
+        self.creation_date = datetime.datetime.now().isoformat()
```

Comment:
You are still defining the creation date as a class property

Reply:
🤦‍♀️